File: app/src/main/java/com/anadi/weatherinfo/Country.java
package com.anadi.weatherinfo;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class Country {
    String name;
    String code;

    public Country(String name, String code) {
        this.name = name;
        this.code = code;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (null == obj)
            return false;

        if (obj instanceof CityInfo) {
            Country other = (Country) obj;
            return other.name.equals(name) &&
                    other.code.equals(code);
        } else if (obj instanceof String) {
            String string = (String) obj;
            return name.equals(string) ||
                    code.equals(string) ||
                    string.equals(this.toString());
        } else
            return false;
    }

    @NonNull
    @Override
    public String toString() {
        return code + " : " + name;
    }
}

File: app/src/main/java/com/anadi/weatherinfo/data/db/location/Country.kt
package com.anadi.weatherinfo.data.db.location

import androidx.room.ColumnInfo

data class Country(
        @ColumnInfo(name = "name")
        val name: String,

        @ColumnInfo(name = "code")
        val code: String) {

    override fun equals(other: Any?): Boolean {
        if (other === this) {
            return true
        }

        return when (other) {
            is Country -> other.name == name && other.code == code
            is String -> name == other || code == other
            else -> false
        }
    }

    override fun hashCode(): Int {
        var result = name.hashCode()
        result = 31 * result + code.hashCode()
        return result
    }

    override fun toString(): String {
        return name
    }

    companion object {
        val EMPTY = Country("", "")
    }
}